src/Symfony/Bridge/Doctrine/Form/Type/EntityType.php

```diff
@@ -35,4 +60,23 @@ public function getName()
     {
         return 'entity';
     }
+    
+    /**
+     * We consider two query builders with an equal SQL string and
+     * equal parameters to be equal.
+     * 
+     * @param QueryBuilder $queryBuilder
+     * 
+     * @return array
+     * 
+     * @internal This method is public to be usable as callback. It should not
+     *           be used in user code.
+     */
+    public function getQueryBuilderPartsForCachingHash($queryBuilder)
```

Comment:
I basically followed how it has already been done here: https://github.com/symfony/symfony/pull/14576/files#diff-0b86d71e1a20e98ac050dd33c4e79373R103